research/format.markdown

```diff
@@ -0,0 +1,124 @@
+| `ip-address`  | SHOULD be internet protocol version 4 address                                                                 | ipv4: A string instance is valid against this attribute if it is a valid representation of an IPv4 address according to the `dotted-quad` ABNF syntax as defined in RFC 2673, section 3.2 [RFC2673] |
+| `ipv6`        | SHOULD be internet protocol version 6 address                                                                 | A string instance is valid against this attribute if it is a valid representation of an IPv6 address as defined in RFC 2373, section 2.2 [RFC2373]                |
+| `host-name`   | This SHOULD be a host-name                                                               | A string instance is valid against this attribute if it is a valid representation for an Internet host name, as defined by RFC 1034, section 3.1 [RFC1034]           |
+
+## Draft 4 to Draft 6
+
+References: [Draft 4](https://json-schema.org/draft-04/draft-fge-json-schema-validation-00#rfc.section.7
+),
+[Draft 6](https://json-schema.org/draft-06/draft-wright-json-schema-validation-01#rfc.section.8)
+
+| Change Type | Format Attribute | Description                                                                                                          |
+|-------------|------------------|----------------------------------------------------------------------------------------------------------------------|
+| Added       | `uri-reference`    | This attribute applies to string instances. A string instance is valid against this attribute if it is a valid URI Reference (either a URI or a relative-reference), according to [RFC3986] |
+| Added       | `uri-template`     | This attribute applies to string instances. A string instance is valid against this attribute if it is a valid URI Template (of any level), according to [RFC6570]                          |
+| Added       | `json-pointer`     | This attribute applies to string instances. A string instance is valid against this attribute if it is a valid JSON Pointer, according to [RFC6901]                                            |
+
+## Draft 6 to Draft 7
+
+References: [Draft 6](https://json-schema.org/draft-06/draft-wright-json-schema-validation-01#rfc.section.8), [Draft 7](https://json-schema.org/draft-07/draft-handrews-json-schema-validation-01#rfc.section.7)
+
+| Change Type | Format Attribute           | Description                                                                                                          |
+|-------------|----------------------------|----------------------------------------------------------------------------------------------------------------------|
+| Added       | `uri-reference`              | This attribute applies to string instances. A string instance is valid against this attribute if it is a valid URI Reference (either a URI or a relative-reference) according to [RFC3986] |
```

Comment:
You already mentioned `uri-reference`, `uri-template`, and `json-pointer` above, so they were not added here, right? Can you go through all the tables and make sure they are not mentioned twice?

Reply:
😅 this happened because I used chat gpt for formatting and it gave combination of both